core/iwasm/include/wasm_export.h

```diff
@@ -1292,6 +1292,32 @@ WASM_RUNTIME_API_EXTERN bool
 wasm_externref_obj2ref(wasm_module_inst_t module_inst,
                        void *extern_obj, uint32_t *p_externref_idx);
 
+/**
+ * Delete external object registered by `wasm_externref_obj2ref`.
+ *
+ * @param module_inst the WASM module instance that the extern object
+ *        belongs to
+ * @param extern_obj the external object to be deleted
+ *
+ * @return true if success, false otherwise
+ */
+WASM_RUNTIME_API_EXTERN bool
+wasm_externref_objdel(wasm_module_inst_t module_inst, void *extern_obj);
+
+/**
+ * Add cleanup callback to release external object.
+ *
+ * @param module_inst the WASM module instance that the extern object
+ *        belongs to
+ * @param extern_obj the external object to which to add the cleanup callback `extern_obj_cleanup`
+ * @param extern_obj_cleanup a callback to release `extern_obj`
+ *
+ * @return true if success, false otherwise
+ */
+WASM_RUNTIME_API_EXTERN bool
+wasm_externref_add_cleanup(wasm_module_inst_t module_inst, void *extern_obj,
```

Comment:
How about `wasm_externref_set_cleanup`? `add_cleanup` may mean there can be multiple cleanups.

Reply:
done! https://github.com/bytecodealliance/wasm-micro-runtime/pull/2455/commits/0a07138e98794e43b5832c115154c7972aa0104c